presto-orc/src/main/java/io/prestosql/orc/writer/TimestampColumnWriter.java

```diff
@@ -261,4 +317,108 @@ public void reset()
         rowGroupColumnStatistics.clear();
         statisticsBuilder = statisticsBuilderSupplier.get();
     }
+
+    private void writeMillis(long millis)
+    {
+        long seconds = millis / MILLISECONDS_PER_SECOND;
+        long millisFraction = floorMod(millis, MILLISECONDS_PER_SECOND);
+        long nanosFraction = millisFraction * NANOSECONDS_PER_MILLISECOND;
+
+        writeValues(millis, seconds, nanosFraction);
+    }
+
+    private void writeValues(long millis, long seconds, long nanosFraction)
```

Comment:
Maybe move this below the other methods.  Also it is a little strange that it has the value in two forms.  I'd consider inlining the stats record in the caller... not sure if that would be better.